src/main/me/doubledutch/options/accessor/AbstractFieldAccessor.java

```diff
@@ -0,0 +1,92 @@
+package me.doubledutch.options.accessor;
+
+import me.doubledutch.options.FieldAccessException;
+import me.doubledutch.options.FieldAccessor;
+import me.doubledutch.options.Result;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+import static me.doubledutch.options.Result.failure;
+import static me.doubledutch.options.Result.success;
+
+public abstract class AbstractFieldAccessor<T> implements FieldAccessor {
```

Comment:
I absolutely agree that this is a much much much more elegant solution! Really like it... the downside is that it goes from a short one file lib, to an actual solution you need to include... but, it might be worth it.